app/services/data_formatter.rb

```diff
@@ -0,0 +1,57 @@
+class DataFormatter
+  def initialize(purchases)
+    @purchases = purchases
+  end
+
+  def category_totals
+    Category.all.map do |c|
+      [c.name, sum_purchases_in_category(c)]
+    end
+  end
+
+  def category_totals_by_month
+    first_purchase = Purchase.all.order(:date).first[:date]
```

Comment:
Is there a way to just get the first purchase back from the DB? Currently you're getting all purchases from the DB, then ordering, then taking the first. It should be possible to write a query that just gets one answer, the first by date.

Reply:
Having had a play around, it turns out you can use .maximum / .minimum on Activerecords and it can order by date as well as returning just that field - much cleaner. Thanks!